### DISCUSSION TITLE
movie update to superbase

### DISCUSSION THREAD

**Discussion by hirockstar** [j471n/j471n.in#21] [Q&A] [answered]
when we update movie to superbase database movie show on local so whenwever we update any data we have to redeploy the app how we can fix that?

**Comment by j471n** [comment]
In my case my movies are showing in the [about page](https://j471n.in/about). And to generate about page I am using MDX, So normally I have to redeploy the page otherwise it will throw error. But if you are not using MDX and you have a general Movies page then you do any of the two things:

1. You can user [revalidate](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration) option in the Next.js
2. You can use supabase webhooks (it will allow you to revalidate the page as you update any data in the table).


I have made a simple guide which you can follow to create a supabase webook. You can check it out [here](https://github.com/j471n/j471n.in#setting-up-the-environment:~:text=REVALIDATE_SECRET%3A%20As%20I,Let%27s%20create%20webhook%3A).

**Reply by hirockstar** [reply]
i create saprate page for that you can check here rock-star.in/movie and i add webhook too but still no respons?

**Reply by j471n** [reply, marked as answer]
you have to create a revalidate route on which supabase will send you a POST request as per my config. And I am also providing the secret key, which you need to store in your production environment variables. 

You can check out my [revalidate API route](https://github.com/j471n/j471n.in/blob/8e4ef4320e8e108bf64edd030a506e409d3b91e8/pages/api/revalidate.ts)

And if you are unable to update the movies data on the website. try to debug the revalidate page, see what is happening when supabase hits the URL. 

first try in the development and then send it to the production.
